### PR TITLE
Adding hover on pins

### PR DESCRIPTION
**Pull request by Feynallein**
Hello, 
I'm working on a web project that uses flutter_map_marker_cluster and adding the hovers on markers using clusters is an important feature of my project. 
If buildPopupOnHover is set to true, a MouseRegion will encapsulates the marker.builder (within the GestureDetector) and the popup will be built when the marker is hovered. When tapping a Marker, only the onMarkerTap method will be called.
I also added two methods : onMarkerHoverEnter and onMarkerHoverExit which follow the spirit of onMarkerTap : they're called respectively on hover enter and on hover exit. 
Finally, I added a timer before the popup is shown, this timer is set to 300 ms by default.

I hope we can do something for the hovering feature to be added as it can be very useful for web projetcs.

Thanks for reading!

### CODE REVIEW

**Comment by Feynallein** [comment]
Up (to remove stale)

**Comment by Feynallein** [comment]
Removing stale

File: pubspec.yaml [hunk 1]
```diff
@@ -11,9 +13,11 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  flutter_map: ^1.0.0
-  flutter_map_marker_popup: ^2.2.0
+  flutter_map: ^2.0.0
   latlong2: ^0.8.1
+  flutter_map_marker_popup:
+    git:
+      url: https://github.com/Feynallein/flutter_map_marker_popup.git
```
Comment (lpongetti) [inline pubspec.yaml:20]: pls dont use git url

**Comment by lpongetti** [comment]
pls fix conflicts

File: pubspec.yaml (hunk 1 above)
Comment (Feynallein) [reply pubspec.yaml]: I had to use this in order to have access to flutter map. Since flutter_map_marker_popup is now updated it's not required anymore.

**Comment by Feynallein** [comment]
Conflicts has been resolved.
Branch has been updated.

**Pull request merged**